FinSearchUnified/ShopwareProcess.php

```diff
@@ -224,6 +193,121 @@ public function getFindologicXml($lang = "de_DE", $start = 0, $length = 0, $save
         return $xmlDocument;
     }
 
+    /**
+     * @param int $productId
+     *
+     * @return null|string
+     */
+    public function getProductById($productId)
+    {
+        $xmlDocument = null;
+        $exporter = Exporter::create(Exporter::TYPE_XML);
+
+        $articlesQuery = $this->articleRepository->createQueryBuilder('articles')
+            ->select('articles')
+            ->where('articles.id = :productId')
+            ->setParameter('productId', $productId);
+        $shopwareArticles = $articlesQuery->getQuery()->execute();
+
+        if (count($shopwareArticles) === 0) {
+            $this->errors[] = 'No article found with given ID';
+            return null;
+        }
+
+        $findologicArticles = [$this->getFindologicArticle($shopwareArticles[0])];
+
+        $xmlArray = new XmlInformation();
+        $xmlArray->total = 1;
+        $xmlArray->count = 1;
+        $xmlArray->items = $findologicArticles;
```

Comment:
Total and count should be the same. The count/total could just be `count($shopwareArticles)`.